release/nightly_tests/dataset/read_parquet_benchmark.py

```diff
@@ -81,6 +81,27 @@ def run_read_parquet_benchmark(benchmark: Benchmark):
                 root=data_dirs[-1],
                 parallelism=1,  # We are testing one task to handle N files
             )
+
+    # Test reading many small files.
+    total_rows = 1024
+    for num_files in [10000, 20000, 50000]:
```

Comment:
Can we pre-generate 50k files instead and put them on S3? That'll test the S3 interaction/metadata fetch etc.